cloudify_hostpool_plugin/tasks.py

```diff
@@ -15,79 +19,113 @@
 
 import os
 import requests
+from requests.exceptions import RequestException, Timeout
 import stat
 
 from cloudify import ctx
-from cloudify.exceptions import NonRecoverableError
+from cloudify.exceptions import NonRecoverableError, RecoverableError
 from cloudify.decorators import operation
 
 
 RUNTIME_PROPERTIES_KEYS = [
-    'ip', 'user', 'port', 'password', 'key', 'host_id', 'public_address'
+    'ip', 'user', 'port',
+    'password', 'key', 'host_id',
+    'public_address'
 ]
 
 
 @operation
-def acquire(service_url, **kwargs):
+def acquire(service_url, **_):
+    '''Acquire a host for the user'''
     ctx.logger.info('Acquire host')
-    response = requests.post('{0}/hosts'.format(service_url))
-    ctx.logger.debug('Response received: {0}'.format(response.text))
-    if response.status_code == requests.codes.CREATED:
-        host = response.json()
-        ctx.logger.info('Acquired host: {0}'.format(host['host']))
-        key_content = host['auth'].get('keyfile')
-        key_path = None
-        if key_content:
-            key_path = _save_keyfile(key_content, host['host_id'])
-        _set_runtime_properties(host, key_path)
-    else:
-        _handle_error(response)
+    try:
+        response = requests.post('%s/hosts' % service_url)
+        ctx.logger.debug('Response received: %s', response.text)
+
+        # pylint: disable=E1101
+        # Harmless warning due to the dynamically-generated LookupDict
+        if response.status_code == requests.codes.CREATED:
+            host = response.json()
+            ctx.logger.info('Acquired host: %s', host['host'])
+
+            key_content = host['auth'].get('keyfile')
+            key_path = None
+            if key_content:
+                key_path = _save_keyfile(key_content, host['host_id'])
+
+            _set_runtime_properties(host, key_path)
+        else:
+            ctx.logger.info('_handle_error(%s)' % response)
+            _handle_error(response)
+    except Timeout as ex:
+        # Catch ConnectTimeout & ReadTimeout errors
+        RecoverableError('Timeout acquiring a host: %s' % ex)
+    except RequestException as ex:
+        # Catastrophic network error
+        ctx.logger.info('RequestException: %s' % ex)
+        NonRecoverableError('Fatal network error acquiring host: %s' % ex)
 
 
 @operation
-def release(service_url, **kwargs):
+def release(service_url, **_):
+    '''Release a host from use by the user'''
     ctx.logger.info('Release host')
     host_id = ctx.instance.runtime_properties['host_id']
-    response = requests.delete('{0}/hosts/{1}'.format(service_url, host_id))
-    ctx.logger.debug('Response received: {0}'.format(str(response)))
-    if not response.ok:
-        _handle_error(response)
-    key_file = ctx.instance.runtime_properties.get('key')
-    if key_file and os.path.exists(key_file):
-        os.unlink(key_file)
-    _delete_runtime_properties()
+    try:
```

Comment:
I prefer this:
```
try:
    response = requests.delete('%s/hosts/%s' % (service_url, host_id))
except Timeout as ex:
    RecoverableError('Timeout acquiring a host: %s' % ex)
except RequestException as ex:
    NonRecoverableError('Fatal network error acquiring host: %s' % ex)
```
because these exceptions only relate to the requests.post method, so it doesn't make sense to include everything in the try block.